### PR TITLE
ConnectEhrStep: Remove polling

### PR DESCRIPTION
**Pull request by bxh**
## Overview

This PR removes polling from ConnectEhrStep, as it was found unnecessary per my conversation with @chrisnowak in #115 .

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

Describe briefly what security risks you considered, why they don't apply, or how they've been mitigated.

### Testing
- [ ] MyDataHelps iOS app
- [ ] MyDataHelps Android app
- [x] [MyDataHelps website](mydatahelps.org)

### Documentation
- [x] I have added relevant Storybook updates to this PR as well.

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

## Reviewers

Assign to the appropriate reviewer(s). Minimally, a second set of eyes is needed ensure no non-public information is published. Consider also including:
- Subject-matter experts
- Style/editing reviewers
- Others requested by the content owner

### CODE REVIEW

File: src/components/container/ProviderSearch/ProviderSearch.tsx [hunk 1]
```diff
@@ -99,10 +99,12 @@ export default function (props: ProviderSearchProps) {
     function connectToProvider(provider: ExternalAccountProvider) {
         const providerID = provider.id;
         if (!props.previewState && !(linkedExternalAccounts[providerID] && linkedExternalAccounts[providerID].status != 'unauthorized')) {
-            MyDataHelps.connectExternalAccount(providerID, { openNewWindow: props.openNewWindow ?? false });
-        }
-        if (props.onProviderSelected) {
-            props.onProviderSelected(provider);
+            MyDataHelps.connectExternalAccount(providerID, { openNewWindow: props.openNewWindow ?? false })
```
[the remark this answers is not in the document]
Comment (bxh) [reply src/components/container/ProviderSearch/ProviderSearch.tsx:102]: Yeah, but I'm doing it on a separate branch. I haven't been able to test all the code paths related to `connectExternalAccount` yet.

File: src/components/step/ConnectEhrStepContainer/ConnectEhrStepContainer.tsx [hunk 2]
```diff
@@ -12,9 +12,11 @@ export default function () {
     const [nextButtonText, setNextButtonText] = useState<string>();
     const [nextButtonDisabled, setNextButtonDisabled] = useState<boolean>(true);
     const [styles, setStyles] = useState<any>({});
-    const [provider, setProvider] = useState<
-        ExternalAccountProvider | undefined
-    >(undefined);
+
+    function onProviderConnected(provider: ExternalAccountProvider) {
+        console.log("Connected to provider ID ", provider.id);
```
Comment (chrisnowak) [inline src/components/step/ConnectEhrStepContainer/ConnectEhrStepContainer.tsx:17]: Remove console log

File: src/components/container/ProviderSearch/ProviderSearch.tsx [hunk 3]
```diff
@@ -99,10 +99,12 @@ export default function (props: ProviderSearchProps) {
...
+                .then(function() {
```
Comment (chrisnowak) [inline src/components/container/ProviderSearch/ProviderSearch.tsx:103]: Is this promise rejected when the user cancels out of the flow or has an error?  Are we sure this only captures "successful connection" scenarios?

Comment (chrisnowak) [reply src/components/container/ProviderSearch/ProviderSearch.tsx:103]: Might try to write a simple WebViewStep in MDH that calls ```MyDataHelps.connectExternalAccount``` to test the various results of the promise across iOS/Android/Web to ensure we have this working 100% and does not allow proceeding if the user cancels, for instance

Comment (bxh) [reply src/components/container/ProviderSearch/ProviderSearch.tsx:103]: > Is this promise rejected when the user cancels out of the flow or has an error? Are we sure this only captures "successful connection" scenarios?

Correct. I've tested it in Long COVID QA. The promise only gets resolved after a user completes all the authorization steps at the 3rd party website. When a participant backs out, e.g., click "Cancel" at Garmin, they will stay at the same survey step.

Comment (bxh) [reply src/components/container/ProviderSearch/ProviderSearch.tsx:103]: https://github.com/CareEvolution/MyDataHelpsUI/pull/116#discussion_r1333546831

**Comment by bxh** [comment]
Also save `providerIDs` in the survey step result. https://github.com/CareEvolution/MyDataHelpsUI/pull/116/commits/180e31ba0fc4e80ca041de30249f891ac3a9b5d4

Can you approve this pull request? @chrisnowak

**Review by chrisnowak — APPROVED** [review]

**Pull request merged**